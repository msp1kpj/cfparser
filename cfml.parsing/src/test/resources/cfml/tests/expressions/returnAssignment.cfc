
component{
public void function test() {
  return foo = bar;
}
}