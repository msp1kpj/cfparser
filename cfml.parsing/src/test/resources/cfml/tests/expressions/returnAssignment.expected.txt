/*===TOKENS===*/
COMPONENT                <component>
'{'                      <{>
PUBLIC                   <public>
IDENTIFIER               <void>
FUNCTION                 <function>
IDENTIFIER               <test>
'('                      <(>
')'                      <)>
'{'                      <{>
RETURN                   <return>
IDENTIFIER               <foo>
'='                      <=>
IDENTIFIER               <bar>
';'                      <;>
'}'                      <}>
'}'                      <}>
/*===TREE===*/
(scriptBlock
  (componentDeclaration
    component
    (componentGuts
      {
      (element
        (functionDeclaration
          (accessType public)
          (typeSpec (multipartIdentifier (identifier void)))
          function
          (identifier test)
          (
          parameterList
          )
          (compoundStatement
            {
            (statement
              (returnStatement
                return
                (assignmentExpression
                  (startExpression
                    (compareExpression
                      (baseExpression (unaryExpression (memberExpression (identifier foo))))
                    )
                  )
                  =
                  (startExpression
                    (compareExpression
                      (baseExpression (unaryExpression (memberExpression (identifier bar))))
                    )
                  )
                )
                ;
              )
            )
            }
          )
        )
      )
      }
    )
  )
)
/*======*/
/*===DECOMPILE===*/
component {
  public void function test()     {
      return foo = bar;

    }
}
/*======*/